lib/managers/KalturaLayoutManager.js

```diff
@@ -696,5 +696,33 @@ class KalturaLayoutManager extends kaltura.KalturaManager {
 		const trackingId = new TrackingIdentifier(type, url, cuePointId, flavorId, seqId);
 		return `tracking/sendBeacon/trackingId/${trackingId.toBase64()}`;
 	}
+
+	static _getPrefix(flavorIds)
+	{
+		const flavors = flavorIds.split(',');
+		let prefix = null;
+		flavors.forEach(function(value){
+			if (value.endsWith('_'))
+				prefix = value;
+		});
+		return prefix;
+	}
+
+	static _getFlavorIds(flavorIdsStr)
+	{
+		const prefix = KalturaLayoutManager._getPrefix(flavorIdsStr);
+		if (!prefix)
+			return flavorIdsStr.split(',');
+
+		//remove the prefix from the list
+		const flavors = flavorIdsStr.split(',');
+		var index = flavors.indexOf(prefix);
+		if (index > -1)
```

Comment:
I think that it is the first one as a general and if not lets agree to throw exception , as consider the case that someone in the future will consider the following as legal 0_,xfftrhr,gdftrh,gfrtfr,1_,hfrufrh,frhfrn
we will try to access the the wrong combination - as a general i think it is safe to assume that the prefix is the first "flavor"